semgrep/tests/e2e/test_ci.py

```diff
@@ -271,208 +256,187 @@ def test_full_run(tmp_path, git_tmp_path_with_commit, snapshot, env, autofix):
-                .replace(base_commit, "<sanitized base_commit>")
-                .replace(__VERSION__, "<sanitized semgrep_version>")
-            )
-            sanitized_output = re.sub(
-                r"python 3\.\d+\.\d+", "python <sanitized_version>", sanitized_output
-            )
-            snapshot.assert_match(sanitized_output, "output.txt")
-
-            # Check correct metadata
-            meta_json = post_mock.mock_calls[0].kwargs["json"]
-            assert meta_json["meta"]["commit"] == head_commit
-            meta_json["meta"]["commit"] = "sanitized"
-            assert meta_json["meta"]["semgrep_version"] == __VERSION__
-            meta_json["meta"]["semgrep_version"] = "<sanitized version>"
-
-            if env.get("GITLAB_CI"):
-                # If in a merge pipeline, base_sha is defined, otherwise is None
-                gitlab_base_sha = (
-                    base_commit
-                    if env.get("CI_MERGE_REQUEST_TARGET_BRANCH_NAME")
-                    else None
-                )
-
-                assert meta_json["meta"]["base_sha"] == gitlab_base_sha
-                meta_json["meta"]["base_sha"] = "sanitized"
-
-            snapshot.assert_match(json.dumps(meta_json, indent=4), "meta.json")
-
-            findings_json = post_mock.mock_calls[1].kwargs["json"]
-            for f in findings_json["findings"]:
-                assert f["commit_date"] is not None
-                f["commit_date"] = "sanitized"
-            snapshot.assert_match(json.dumps(findings_json, indent=4), "findings.json")
-
-            ignores_json = post_mock.mock_calls[2].kwargs["json"]
-            for f in ignores_json["findings"]:
-                assert f["commit_date"] is not None
-                f["commit_date"] = "sanitized"
-            snapshot.assert_match(json.dumps(ignores_json, indent=4), "ignores.json")
-
-            complete_json = post_mock.mock_calls[3].kwargs["json"]
-            complete_json["stats"]["total_time"] = 0.5  # Sanitize time for comparison
-            snapshot.assert_match(json.dumps(complete_json, indent=4), "complete.json")
+    runner = CliRunner(
+        env={
+            **env,
+            SEMGREP_SETTING_ENVVAR_NAME: str(tmp_path),
+            auth.SEMGREP_LOGIN_TOKEN_ENVVAR_NAME: "fake_key",
+        }
+    )
+    result = runner.invoke(cli, ["ci"], env={})
+
+    # Remove commit hashes from output
+    sanitized_output = (
+        result.output.replace(head_commit, "<sanitized head_commit>")
+        .replace(head_commit[:7], "<sanitized head_commit>")
+        .replace(base_commit, "<sanitized base_commit>")
+        .replace(__VERSION__, "<sanitized semgrep_version>")
+    )
+    sanitized_output = re.sub(
+        r"python 3\.\d+\.\d+", "python <sanitized_version>", sanitized_output
+    )
+    snapshot.assert_match(sanitized_output, "output.txt")
 
+    post_calls = AppSession.post.call_args_list  # type: ignore
```

Comment:
1. Instead of keeping the return value of the mock call around as a variable, We can directly access the mocked object under the `semgrep.` namespace.
2. Instead of checking `.mock_calls`, which contains all attribute access, etc. (this is why we had that comment saying `Return a different mock so post_mock only contains direct calls to session.post`), we can access `.call_args_list` which contains only the direct calls.
3. This should work on python 3.7 according to docs, so I removed the `skipif` decorator.
4. The caveat here is that mypy doesn't like us accessing the attributes that mock added. I didn't find a nice solution yet.